packages/keycloak-user-management/src/components/forms/UserForm/utils.tsx

```diff
@@ -75,60 +132,63 @@ export const submitForm = async (
     practitionerIsEditMode,
   } = getUserFormPayload(values);
 
-  let keycloakUserPromise: Promise<string | void>;
+  /**
+   * callback to update groups and practitioners upon successfully updating keycloak user
+   *
+   * @param keycloakUserId - id of the keycloak user
+   */
+  const updateGroupsAndPractitioner = async (keycloakUserId: string) => {
```

Comment:
we should maybe extract this function. It's unclear how the dependencies are flowing (some passed as args and others accessing directly)

Reply:
This is a callback function, I cannot just add extra parameters to it, since then I would also have to modify its callpoint and find a way to make those arguments available there. 
thought using closures here would avoid extra abstractions.